### PR TITLE
4826: Get orders api call

### PR DESCRIPTION
**Pull request by kat1906**
Story: [AB#4826](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/4826)
Task: [AB#6756](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/6756) [AB#6818](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/6818)

### CODE REVIEW

File: app/helpers/dateFormatter.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { formatDate } from './dateFormatter';
+
```
Comment (kat1906) [inline app/helpers/dateFormatter.test.js:2]: https://www.valentinog.com/blog/datetime/#tolocaledatestring-or-intldatetimeformat

File: app/config.js [hunk 2]
```diff
@@ -45,7 +48,7 @@ module.exports = {
   redisUrl: process.env.REDIS_URL || 'localhost',
 
   // The port redis is running
-  redisPort: process.env.REDIS_PORT || 6379,
+  redisPort: process.env.REDIS_PORT || 6380,
```
Comment (alibombaye) [inline app/config.js:51]: Just a reminder when you get round to it

File: app/helpers/dateFormatter.test.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+describe('formatDate', () => {
+  it('returns correctly formatted date', () => {
+    expect(formatDate('2020-05-06T11:29:52.4965647Z')).toEqual('6 May 2020');
+    expect(formatDate('2025-12-31T11:29:52.4965647Z')).toEqual('31 December 2025');
```
Comment (alibombaye) [inline app/helpers/dateFormatter.test.js:6]: sweet

File: app/config.js (hunk 2 above)
Comment (kat1906) [reply app/config.js:51]: lol every time 🤦

File: app/pages/dashboard/contextCreator.js [hunk 4]
```diff
@@ -1,9 +1,43 @@
...
+        data: order.lastUpdatedBy,
+        dataTestId: `lastUpdatedBy-${order.orderId}`,
+      },
+      {
+        data: formatDate(order.lastUpdated),
+        dataTestId: `lastUpdated-${order.orderId}`,
+      },
+      {
+        data: formatDate(order.dateCreated),
+        dataTestId: `dateCreated-${order.orderId}`,
```
Comment (alibombaye) [inline app/pages/dashboard/contextCreator.js:27]: really picky but I would flip them so {{ orderId }}-{{ column name }}

File: app/pages/dashboard/contextCreator.test.js [hunk 5]
```diff
@@ -29,4 +48,75 @@ describe('getContext', () => {
...
+//     if (order.status === 'Submitted') acc.submittedOrders.push(formattedOrder);
+//     else acc.unsubmittedOrders.push(formattedOrder);
+//     return acc;
+//   }, { submittedOrders: [], unsubmittedOrders: [] });
+
+//   return ({
+//     submittedOrders: formattedOrders.submittedOrders,
+//     unsubmittedOrders: formattedOrders.unsubmittedOrders,
+//   });
+// };
```
Comment (alibombaye) [inline app/pages/dashboard/contextCreator.test.js:86]: take it you'll remove this later?

File: app/pages/dashboard/contextCreator.test.js [hunk 6]
```diff
@@ -29,4 +48,75 @@ describe('getContext', () => {
...
+  fdescribe('ordersData', () => {
```
Comment (alibombaye) [inline app/pages/dashboard/contextCreator.test.js:87]: got an `f` here

File: app/pages/dashboard/contextCreator.test.js [hunk 7]
```diff
@@ -29,4 +48,75 @@ describe('getContext', () => {
...
+      expect(unsubmittedOrder1[1].data).toEqual('Some Order');
+      expect(unsubmittedOrder1[1].dataTestId).toEqual('description-order1');
+      expect(unsubmittedOrder1[2].data).toEqual('Bob Smith');
+      expect(unsubmittedOrder1[2].dataTestId).toEqual('lastUpdatedBy-order1');
+      expect(unsubmittedOrder1[3].data).toEqual('6 May 2020');
+      expect(unsubmittedOrder1[3].dataTestId).toEqual('lastUpdated-order1');
+      expect(unsubmittedOrder1[4].data).toEqual('6 May 2020');
+      expect(unsubmittedOrder1[4].dataTestId).toEqual('dateCreated-order1');
+    });
+  });
```
Comment (alibombaye) [inline app/pages/dashboard/contextCreator.test.js:121]: I'd add one more test case with multiple submitted and unsubmitted orders but just check the length of these are what you expect

File: app/pages/dashboard/controller.js [hunk 8]
```diff
@@ -1,3 +1,30 @@
...
+//   },
+//   {
+//     orderId: 'C000012-01',
+//     orderDescription: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean m',
+//     lastUpdatedBy: 'Bobbybobbybobby Smithsmith-Smithsmith-Smithsmith',
+//     lastUpdated: '2020-12-09T09:29:52.49657Z',
+//     dateCreated: '2020-10-09T09:29:52.4965701Z',
+//     status: 'Submitted',
+//   },
+// ];
```
Comment (alibombaye) [inline app/pages/dashboard/controller.js:23]: comment here and below to remove?

File: app/pages/dashboard/controller.test.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    orderDescription: 'Some Order',
+    lastUpdatedBy: 'Alice Smith',
+    lastUpdated: '2020-05-06T11:29:52.4965647Z',
+    dateCreated: '2020-01-06T09:29:52.4965653Z',
+    status: 'Unsubmitted',
+  },
+  {
+    orderId: 'C000012-01',
+    orderDescription: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean m',
+    lastUpdatedBy: 'Bobbybobbybobby Smithsmith-Smithsmith-Smithsmith',
```
Comment (alibombaye) [inline app/pages/dashboard/controller.test.js:25]: maybe change this dummy data 😂

File: app/pages/dashboard/template.test.js [hunk 10]
```diff
@@ -91,5 +123,55 @@ describe('dashboard page', () => {
...
+        expect(lastUpdatedBy1.length).toEqual(1);
+        expect(lastUpdatedBy1.text().trim()).toEqual('Alice Smith');
+        expect(lastUpdated1.length).toEqual(1);
+        expect(lastUpdated1.text().trim()).toEqual('9 June 2021');
+        expect(dateCreated1.length).toEqual(1);
+        expect(dateCreated1.text().trim()).toEqual('9 October 2020');
+
+        expect(table.length).toEqual(1);
+        expect(table.find('[data-test-id="table"]').length).toEqual(1);
+        expect(table.find('[data-test-id="table-headings"]').length).toEqual(1);
```
Comment (alibombaye) [inline app/pages/dashboard/template.test.js:161]: Don't need to check the above 3 lines again.

**Pull request merged**